src/services/paras/ParasService.spec.ts

```diff
@@ -516,33 +538,60 @@ describe('ParasService', () => {
 			expect(sanitizeNumbers(response)).toMatchObject(expectedResponse);
 		});
 
-		/**
-		 * The goal of this test is to manipulate the number of the finalized block so that it is less than
-		 * the expected `finishHead`, but higher the `beginEnd` which would denote we are in the `endPeriod` phase
-		 * of the current auction.
-		 */
-		it('Should return the correct `ending` phase', async () => {
-			const overrideHeader = {
-				parentHash:
-					'0x3d489d71f8fd2e15259df5059a1497436e6b73497500a303b1a705993e25cb27',
-				number: 20000,
-				stateRoot:
-					'0xa0089595e48850a8a00081dd987a4735d0e8f94ac98af89030521f23f6cb8e31',
-				extrinsicsRoot:
-					'0x2d5d3fdb96b487d480b08b64ed69a65433c1713ae3579dd23704cb790aa3b2ae',
-				digest: {},
+		it('Should return the correct data during a startPeriod phase', async () => {
+			const overrideHeader = generateOverrideHeader(770000);
+			const header = polkadotRegistry.createType('Header', overrideHeader);
+
+			// Override the mockApi
+			(mockApi.rpc.chain.getHeader as unknown) = () =>
+				Promise.resolve().then(() => header);
```

Comment:
Just a general thought; I wonder whether there's a nice way to avoid needing to remember to "reset" this sort of thing at the end of the call. Eg with immer you might do something like:

```
const thisMockApi = immer(mockApi, mockApi => {
    (mockApi.rpc.chain.getHeader as unknown) = () =>
        Promise.resolve().then(() => header);
});
```

And then the original `mockApi` would be untouched. Just thinking out loud really :)

Reply:
Yea I am definitely open to this, as it's an unfortunate pattern we have in the testing and I think this could definitely help and improve it. If we do use this I think it would be useful to do a seperate PR that overhauls all the times where we change the mockApi then reestablish it.  This is super apparent in `/blocks` tests.

Reply:
TIL about `immer` btw. Thank you :)

Reply:
Yep agreed; a separate PR would def make more sense!

(Ah I love immer :D)